rclrs/src/node/mod.rs

```diff
@@ -38,18 +38,33 @@ impl Drop for NodeHandle {
     }
 }
 
+/// A processing unit that can communicate with other nodes.
+///
+/// Nodes are a core concept in ROS 2. Refer to the official ["Understanding ROS 2 nodes"][1]
+/// tutorial for an introduction.
+///
+/// Ownership of the node is shared with all [`Publisher`]s and [`Subscription`]s created from it.
+/// That means that even after the node itself is dropped, it will continue to exist and be
+/// displayed by e.g. `ros2 topic` as long as its publishers and subscriptions are not dropped.
+///
+/// [1]: https://docs.ros.org/en/foxy/Tutorials/Understanding-ROS2-Nodes.html
```

Comment:
I'd link the code documentation, not the tutorials

Reply:
Also, this links to `foxy`, this would need to be updated every time there's a new release

Reply:
Which specific code documentation? I like the tutorial because it explains the concept in a language-agnostic manner. We can link to the Rolling version to avoid the staleness problem: https://docs.ros.org/en/rolling/Tutorials/Understanding-ROS2-Nodes.html.

Reply:
I changed the link to the Rolling version.